app/src/main/java/org/wikipedia/history/HistoryViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package org.wikipedia.history
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.wikipedia.database.AppDatabase
+import org.wikipedia.util.Resource
+
+class HistoryViewModel : ViewModel() {
+
+    private val handler = CoroutineExceptionHandler { _, throwable ->
+        _uiState.value = Resource.Error(throwable)
+    }
+
+    var searchQuery: String? = null
+
+    private val _uiState = MutableStateFlow(Resource<List<Any>>())
+    val uiState = _uiState.asStateFlow()
+
+    private val _actionUiState = MutableStateFlow(Resource<Boolean>())
+    val actionUiState = _actionUiState.asStateFlow()
+
+    init {
+        reloadHistoryItems()
+    }
+
+    fun reloadHistoryItems() {
+        viewModelScope.launch(handler) {
+            loadHistoryItems()
+        }
+    }
+
+    private suspend fun loadHistoryItems() {
+        withContext(Dispatchers.IO) {
```

Comment:
I would move this Dispatcher.IO to the repository and wrap this code with viewModelScope.launch(handler). Then we could remove it from the reloadHistoryItems (even more, we could remove the realodHistoryItems fun), and remove suspend to maintain consistence with the another ViewModel methods:

```suggestion
        private fun loadHistoryItems() {
        viewModelScope.launch(handler) {
            val items = historyRepository.filterHistoryItems(searchQuery.orEmpty())
            _uiState.value = Resource.Success(items)
        }
    }
```